### PR TITLE
[GH-2725] Implement GeoSeries: type, unary_union, delaunay_triangles, voronoi_polygons, disjoint, m

### LINKED ISSUE #2725: Implement GeoSeries: type, unary_union, delaunay_triangles, voronoi_polygons, disjoint, m
### Description

Implement Tier 2 batch D GeoSeries functions for the geopandas compatibility module.

**Functions to implement:**
- `type` — property returning geometry type names (alias for `geom_type`)
- `unary_union` — deprecated property that delegates to `union_all()`
- `delaunay_triangles` — element-wise Delaunay triangulation via `ST_DelaunayTriangles`
- `voronoi_polygons` — element-wise Voronoi diagram via `ST_VoronoiPolygons`
- `disjoint` — binary predicate via `ST_Disjoint`
- `m` — returns M coordinate via `ST_M`

**Not included (blocked by upstream bugs):**
- `geom_equals` — blocked by #2722 (`ST_Equals` crashes on `GeometryCollection`)
- `frechet_distance` — blocked by #2720 (`ST_FrechetDistance` returns 0.0 for empty geometries)
- `hausdorff_distance` — blocked by #2721 (`ST_HausdorffDistance` returns 0.0 for empty geometries)

Part of #2230.

### PR DESCRIPTION
**Pull request by jiayuasu**
## Did you read the Contributor Guide?
- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?
- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #2725

## What changes were proposed in this PR?

Implement 6 GeoSeries functions for the geopandas compatibility module:

**General methods and attributes:**
- `type` (property) — alias for `geom_type`, returns geometry type names as a Series
- `m` (property) -> `ST_M` — returns the M coordinate of point geometries

**Constructive methods:**
- `delaunay_triangles(tolerance, only_edges)` -> `ST_DelaunayTriangles`
- `voronoi_polygons(tolerance, extend_to, only_edges)` -> `ST_VoronoiPolygons`

**Binary predicates:**
- `disjoint(other, align)` -> `ST_Disjoint`

**Aggregation:**
- `unary_union` (property) — deprecated, delegates to `union_all()` with `FutureWarning`

Also fixes `GeoDataFrame.type` to delegate to `geom_type` instead of raising `NotImplementedError`.

Each function follows the established pattern from #2701 and #2710:
- **base.py**: Docstring with examples + `_delegate_to_geometry_column` call
- **geoseries.py**: ST function call via `_query_geometry_column` or `_row_wise_operation`
- **test_geoseries.py**: Unit test with expected values + GeoDataFrame delegation check
- **test_match_geopandas_series.py**: Comparison test against real geopandas output

**Semantic differences documented:**
- `delaunay_triangles` and `voronoi_polygons` operate **per row** (each input geometry produces a `GeometryCollection`), unlike geopandas which collects all vertices across the GeoSeries and computes a single result. This is noted in the docstrings.

**Not included (blocked by upstream Sedona bugs):**
- `geom_equals` — #2722 (`ST_Equals` crashes on `GeometryCollection` inputs)
- `frechet_distance` — #2720 (`ST_FrechetDistance` returns 0.0 for empty geometries)
- `hausdorff_distance` — #2721 (`ST_HausdorffDistance` returns 0.0 for empty geometries)

## How was this patch tested?

12 new tests (6 unit tests + 6 match tests) all passing locally.

## Did this PR include necessary documentation updates?

- No, this PR does not affect any public API so no need to change the documentation.

### CODE REVIEW

File: python/tests/geopandas/test_geoseries.py [hunk 1]
```diff
@@ -2220,6 +2330,15 @@ def test_distance(self):
...
 
+    def test_frechet_distance(self):
+        pass
+
+    def test_hausdorff_distance(self):
+        pass
+
+    def test_geom_equals(self):
+        pass
+
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/tests/geopandas/test_geoseries.py]: Removed the stub in 546297d.

File: python/tests/geopandas/test_geoseries.py [hunk 2]
```diff
@@ -2220,6 +2330,15 @@ def test_distance(self):
...
         self.check_pd_series_equal(df_result, expected)
 
+    def test_frechet_distance(self):
+        pass
+
+    def test_hausdorff_distance(self):
+        pass
+
+    def test_geom_equals(self):
+        pass
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/tests/geopandas/test_geoseries.py]: Removed the stubs in 546297d.

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 3]
```diff
@@ -1214,6 +1254,15 @@ def test_distance(self):
                 )
                 self.check_pd_series_equal(sgpd_result, gpd_result)
 
+    def test_frechet_distance(self):
+        pass
+
+    def test_hausdorff_distance(self):
+        pass
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/tests/geopandas/test_match_geopandas_series.py]: Removed the stubs in 546297d.

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 4]
```diff
@@ -1214,6 +1254,15 @@ def test_distance(self):
...
+
+    def test_geom_equals(self):
+        pass
+
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/tests/geopandas/test_match_geopandas_series.py]: Removed the stub in 546297d.

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 5]
```diff
@@ -977,7 +999,15 @@ def test_line_merge(self):
...
     def test_unary_union(self):
-        pass
+        lst = [g for geom in self.geoms for g in geom if g.is_valid]
+        with pytest.warns(FutureWarning, match="unary_union"):
+            sgpd_result = GeoSeries(lst).unary_union
+        import warnings
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", FutureWarning)
+            gpd_result = gpd.GeoSeries(lst).unary_union
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/tests/geopandas/test_match_geopandas_series.py]: Moved to module scope in 546297d.

File: python/sedona/spark/geopandas/geoseries.py [hunk 6]
```diff
@@ -984,12 +981,24 @@ def convex_hull(self) -> "GeoSeries":
...
+        )
 
     def voronoi_polygons(self, tolerance=0.0, extend_to=None, only_edges=False):
-        # Implementation of the abstract method.
-        raise NotImplementedError("This method is not implemented yet.")
+        if only_edges:
+            raise NotImplementedError(
+                "Sedona does not support only_edges=True for voronoi_polygons."
+            )
+        spark_expr = stf.ST_VoronoiPolygons(self.spark.column, tolerance, extend_to)
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/geoseries.py:1001]: Now raises `NotImplementedError` when `extend_to is not None`. Fixed in 546297d.

File: python/sedona/spark/geopandas/base.py [hunk 7]
```diff
@@ -772,11 +794,80 @@ def convex_hull(self):
...
+            one ``GeometryCollection`` containing its Voronoi polygons.
+            The output GeoSeries has the same length as the input.
+
+        Parameters
+        ----------
+        tolerance : float, default 0.0
+            Snapping tolerance for vertices to be considered equal.
+        extend_to : Geometry, default None
+            Not supported in Sedona.
+        only_edges : bool, default False
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/base.py:852]: Updated docstring to clarify that passing a non-None value raises `NotImplementedError`. Fixed in 546297d.

File: python/sedona/spark/geopandas/base.py [hunk 8]
```diff
@@ -1337,9 +1429,35 @@ def line_merge(self, directed=False):
...
-    # def unary_union(self):
-    #     raise NotImplementedError("This method is not implemented yet.")
+    @property
+    def unary_union(self):
+        """Returns a geometry containing the union of all geometries in the
+        ``GeoSeries``.
+
+        .. deprecated::
+            The ``unary_union`` attribute is deprecated. Use
+            :meth:`union_all` instead.
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/base.py]: Fixed in 389ec26. Replaced `.. deprecated::` directive with plain text to avoid Sphinx version argument requirement.

File: python/sedona/spark/geopandas/base.py [hunk 9]
```diff
@@ -1502,6 +1620,36 @@ def crosses(self, other, align=None) -> ps.Series:
         """
         return _delegate_to_geometry_column("crosses", self, other, align)
 
+    def disjoint(self, other, align=None):
+        """Returns a ``Series`` of ``dtype('bool')`` with value ``True`` for
+        each aligned geometry that is disjoint from `other`.
+
+        An object is said to be disjoint from `other` if its
+        `boundary` and `interior` does not intersect at all with those of the
+        other.
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/base.py]: Fixed in 389ec26.

File: python/sedona/spark/geopandas/geoseries.py [hunk 10]
```diff
@@ -1203,8 +1222,17 @@ def crosses(self, other, align=None) -> pspd.Series:
         return _to_bool(result)
 
     def disjoint(self, other, align=None):
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/geoseries.py]: Fixed in 389ec26.

File: python/sedona/spark/geopandas/geoseries.py [hunk 11]
```diff
@@ -984,12 +981,28 @@ def convex_hull(self) -> "GeoSeries":
         )
 
     def delaunay_triangles(self, tolerance=0.0, only_edges=False):
-        # Implementation of the abstract method.
-        raise NotImplementedError("This method is not implemented yet.")
+        spark_expr = stf.ST_DelaunayTriangles(
+            self.spark.column, tolerance, int(only_edges)
+        )
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/geoseries.py:986]: The existing tests already cover the default path. Adding edge-case tests for `only_edges=True` is out of scope for this PR.

File: python/sedona/spark/geopandas/geoseries.py [hunk 12]
```diff
@@ -984,12 +981,28 @@ def convex_hull(self) -> "GeoSeries":
...
-        # Implementation of the abstract method.
-        raise NotImplementedError("This method is not implemented yet.")
+        if only_edges:
+            raise NotImplementedError(
+                "Sedona does not support only_edges=True for voronoi_polygons."
+            )
+        if extend_to is not None:
+            raise NotImplementedError(
+                "Sedona does not support extend_to for voronoi_polygons."
+            )
```
[the remark this answers is not in the document]
Comment (jiayuasu) [reply python/sedona/spark/geopandas/geoseries.py:1000]: The `voronoi_polygons(only_edges=True)` test already exists in `test_voronoi_polygons`. The `extend_to` guard follows the same pattern — out of scope for additional tests in this PR.
